addon/validations/factory.js

```diff
@@ -207,21 +207,21 @@ function createValidationsClass(inheritedValidationsClass, validations = {}) {
     return obj;
   }, validationRules);
 
-  // Create the CPs that will be a part of the `attrs` object
-  const attrCPs = validatableAttributes.reduce((obj, attribute) => {
-    assign(obj, attribute, createCPValidationFor(attribute, get(validationRules, attribute)), true);
-    return obj;
-  }, {});
-
   // Create the mixin that holds all the top level validation props (isValid, messages, etc)
   const TopLevelProps = createTopLevelPropsMixin(validatableAttributes);
 
   // Create the `attrs` class which will add the current model reference once instantiated
-  const Attrs = Ember.Object.extend(attrCPs, {
+  const Attrs = Ember.Object.extend({
     init() {
       this._super(...arguments);
       const model = this.get('_model');
 
+      // Create the CPs
+      validatableAttributes.forEach(attribute => {
+        const cp = createCPValidationFor(attribute, get(validationRules, attribute), owner);
+        assign(this, attribute, cp, true);
```

Comment:
Oh man, I've been gone for a week and I already forgot we spoke about this. I feel like something like this will be very handy so we should def. explore a more optimal path but this is already being done in the current version of this library via the `assign` util method. When you have nested keys, it creates a new Ember.Object for each nested key, and at the end, uses `defineProperty` on the instance. See [here](https://github.com/offirgolan/ember-cp-validations/blob/master/addon/utils/assign.js#L37).

If you end up with a viable solution, im assuming it would only work for the latest ember version and above. Would we be able to back port such a feature?